lib/puppet/util/rpm_compare.rb

```diff
@@ -0,0 +1,206 @@
+module Puppet::Util::RpmCompare
+
+  ARCH_LIST = %w(
+    noarch
+    i386
+    i686
+    ppc
+    ppc64
+    armv3l
+    armv4b
+    armv4l
+    armv4tl
+    armv5tel
+    armv5tejl
+    armv6l
+    armv7l
+    m68kmint
+    s390
+    s390x
+    ia64
+    x86_64
+    sh3
+    sh4
+  ).freeze
```

Comment:
nit: this can be shorten as:
```ruby

ARCH_LIST = %w(
  noarch i386 i686 ppc ppc64 armv3l armv4b armv4l armv4tl armv5tel
  armv5tejl armv6l armv7l m68kmint s390 s390x ia64 x86_64 sh3 sh4
).freeze
```